### DISCUSSION TITLE
Bumping single pkg only with Lerna

### DISCUSSION THREAD

**Discussion by shay-vanti** [lerna/lerna#3451] [Q&A] [answered]
Hey all. I've been using Lerna to create a monorepo with 2 npm packages; lets call them A, B.

Now, when I commit a change on package A, and then publish, Lerna bumps both A and B.
I'm running `lerna publish --registry=https://npm.pkg.github.com/` and Lerna asks me if I want to bump all pkgs (A+B) but doesn't let me choose one. It is either all or nothing.

I feel like I missed something in the docs. Can you suggest how should I only bump A or B?

Thanks.

**Comment by ghiscoding** [comment, marked as answer]
Lerna has 2 version modes and that used to be mentioned in the readme but it was removed (I guess they now mention it on their website). 
- version mode will bump all packages to the same version
- independent mode will prompt you for a new version on every package that it detects a change

in `lerna.json` you can configure one or the other

#### independent mode
```js
{ 
  "version": "independent"
}
```

#### version mode will bump to the next conventional version found when `conventionalCommits` is enabled, for example if a commit with the word "feat" is found, it will bump all packages to a `minor` which would be `1.1.0` in this case. I'm not sure what would happen when conventional commit isn't enabled, I always use it. 
```js
{ 
  "version": "1.0.0",
  "command": {
    "conventionalCommits": true
  }
}
```
there's also ignore changes that you could add as well to avoid pushing no version for unrelated changes like docs
```js
{ 
  "version": "1.0.0",
  "command": {
    "conventionalCommits": true
  }.
  "ignoreChanges": [
    "**/*.md"
  ]
}
```
if that's not related to what you're asking for, then you need to be a bit more precise in your question. I would suggest to watch this great YouTube video [How to Use Lerna](https://www.youtube.com/watch?v=p6qoJ4apCjA), there also other and newer videos on Lerna's website

**Reply by shay-vanti** [reply]
@ghiscoding First of all, thanks. That independent config was almost a complete solution to my answer.
I've tested it and there's an additional limitation I've facing. So when in independent mode, I can set the version bump for each and every pkg under Lerna individually. However, I can't just "skip" a bump. If I have 2 pkgs: A(1.0.6) and B (1.0.6), and only made changes to A, I would expect to be able to bump A to 1.0.7, while leaving B intact. Simply the same as before (1.0.6).. while Lerna's  interactive CLI forces me to choose a version.
Even if I choose the same Custom Version for B (explicitly typing 1.0.3) Lerna is angry at me.

So as a following question to your answer, how do I (can I) leave a pkg un-bumped on a `Lerna --publish`?
Thanks!

**Reply by ghiscoding** [reply]
sorry I'm not sure of the options to use for that since I prefer not to use independent mode (I prefer to have all packages on the same version), I know if you have `conventionalCommits` enabled, it will always try to bump even when there's no change and it's because of a 3rd party lib of conventional commits (I'm not sure if it's your use case). You should take a look at all available options for the `lerna version`, see [docs](https://github.com/lerna/lerna/tree/main/commands/version)

**Reply by shay-vanti** [reply]
[SOLVED - EDITED]

It looks like I was missing the point. Indeed the `lerna.json` version change to `independent` solved my problem. I was probably doing it wrong beforehand. Now the versioning is working as expected.
